src/Domain/Entity/ProgrammeItem.php

```diff
@@ -135,4 +140,19 @@ public function isVideo(): bool
     {
         return ($this->mediaType === MediaTypeEnum::VIDEO);
     }
+
+    public function getDownloadableMediaSets(): array
+    {
+        return $this->downloadableMediaSets;
+    }
+
+    public function isDownloadable(): bool
+    {
+        return !empty($this->downloadableMediaSets);
+    }
+
+    public function hasFutureAvailability(): bool
+    {
+        return !$this->isStreamable() && $this->getStreamableFrom();
```

Comment:
What if the getStreamableFrom() is in the past?

Reply:
Just added a comment explaining that.